packages/react-components/src/components/Picker/Trigger.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import cx from 'clsx';
+import {
+  ChevronDown,
+  ChevronUp,
+  Close,
+} from '@livechat/design-system-icons/react/material';
+import { Icon, IconSizeName, IconTypeName } from '../Icon';
+import styles from './Trigger.module.scss';
+import { KeyCodes } from '../../utils/keyCodes';
+
+const baseClass = 'picker-trigger';
+
+export type TriggerSize = 'compact' | 'medium' | 'large';
+
+export interface ITriggerProps {
+  isDisabled?: boolean;
+  isError?: boolean;
+  isItemSelected: boolean;
+  isOpen: boolean;
+  isRequired?: boolean;
+  size?: TriggerSize;
+  onClick: () => void;
+  onClearClick: () => void;
+  onFilter: (text: string) => void;
+}
+
+export const Trigger: React.FC<ITriggerProps> = ({
+  children,
+  isDisabled,
+  isError,
+  isItemSelected,
+  isOpen,
+  isRequired,
+  size = 'medium',
+  onClick,
+  onClearClick,
+  onFilter,
+}) => {
+  const mergedClassNames = cx(
+    styles[baseClass],
+    styles[`${baseClass}--${size}`],
+    isDisabled && styles[`${baseClass}--disabled`],
+    isOpen && styles[`${baseClass}--focused`],
+    isError && styles[`${baseClass}--error`]
+  );
+
+  React.useEffect(() => {
+    const onKeyDown = (e: KeyboardEvent) => {
+      const el = document.getElementById('picker-trigger');
+      const isFocused = document.activeElement === el;
```

Comment:
Instead of quering an element from DOM by it's id, the handler can use a React's ref instead
```suggestion
      const isFocused = document.activeElement === pickerTriggerRef.current;
```